src/toClass.js

```diff
@@ -0,0 +1,29 @@
+import { Component } from 'react';
+import isReactClass from './isReactClass';
+import compose from './compose';
+import getDisplayName from './getDisplayName';
+import setDisplayName from './setDisplayName';
+import setContextTypes from './setContextTypes';
+import setDefaultProps from './setDefaultProps';
+import setPropTypes from './setPropTypes';
+
+const toClass = baseComponent => {
+  if (isReactClass(baseComponent)) {
+    return baseComponent;
+  }
+
+  const ClassWrapper = class extends Component {
+    render() {
+      return baseComponent(this.props, this.context);
+    }
+  };
+
+  return compose(
+    setDisplayName(getDisplayName(baseComponent)),
+    setContextTypes(baseComponent.contextTypes),
+    setDefaultProps(baseComponent.defaultProps),
```

Comment:
I'd prefer not to include a `setDefaultProps()` and direct people toward either `defaultProps()` or `setStatic('defaultProps')` instead.